src/irc/core/modes.c

```diff
@@ -748,6 +748,7 @@ static char *get_nicks(IRC_SERVER_REC *server, WI_ITEM_REC *item,
 				signal_emit("error command", 1,
 					    GINT_TO_POINTER(CMDERR_NOT_GOOD_IDEA));
 				signal_stop();
+                                g_strfreev(matches);
```

Comment:
This would be better placed after the `g_string_free` so that it matches the final sequence and the allocation order, other than this small nit the patch looks good to me.